docs/spec/v1beta1/provider.md

```diff
@@ -730,3 +732,134 @@ kubectl create secret generic webhook-url \
+
+The request includes the `event_type` and `client_payload` fields:
+
+* The `event_type` is generated by GitHub Dispatch provider by combining the Kind, Name and Namespace of the involved object in the format `{Kind}/{Name}.{Namespace}`. For example, the `event_type` for Kustomization resource named `podinfo` in the `default` namespace looks like this: `Kustomization/podinfo.default`
+
+* The `client_payload` contains the event data from the notification controller and looks like this:
+
+```yaml
+{
+  involvedObject: {
+    apiVersion: kustomize.toolkit.fluxcd.io/v1beta2,
+    kind: Kustomization,
+    name: podinfo,
+    namespace: flux-system,
+    resourceVersion: 426573,
+    uid: b9b8554d-be26-4a3d-a97f-65f3276a097a
+  },
+  message: Deployment/podinfo/podinfo configured,
+  metadata: {
+    revision: main/96139968ca46b53462d1bf94de410a811d2026a1
+  },
+  reason: Progressing,
+  reportingController: kustomize-controller,
+  reportingInstance: kustomize-controller-79464d8dc5-nb9c4,
+  severity: info,
+  timestamp: 2022-04-20T12:20:28Z
+}
+```
+
+#### Setting up a GitHub workflow
+
+You need to add `repository_dispatch` trigger to trigger your GitHub workflow. You can use a wildcard for `types` value to use only the required information from the `event_type` value in the webhook. For example, you can use only Kustomization name:
+
+```yaml
+name: test-github-dispatch-provider
+on:
+  repository_dispatch:
+    types:
+      - "*podinfo*"
+```
+
+Assuming that we deploy all Flux kustomization resources in the same namespace, it will be useful to have a unique kustomization resource name for each application in each environment. This will allow you to use only `event_type` to trigger tests for a exact application in the desired environment.
+
+Let's say we have following folder structure for applications kustomization manifests:
+
+```bash
+app1/
+    overlays/
+        staging/
+        production/
+app2/
+    overlays/
+        staging/
+        production/
+```
+
+You can then create a flux kustomization resource for each overlay to have unique `event_type` looks like this, for example, for app1 and the staging overlay: `Kustomization/app1-staging.default`.
+
+The kustomization manifest for app1/staging:
+
+```yaml
+apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
+kind: Kustomization
+metadata:
+  name: app1-staging
+  namespace: default
+  spec:
+    path: "./app1/staging"
+```
+
+You can the trigger tests for app1 in a staging environment when the app1 resources defined in `./app1/staging/` are configured by Flux:
+
+```yaml
+name: test-github-dispatch-provider
+on:
+  repository_dispatch:
+    types:
+      - "*app1-staging*"
+```
+
+If the information from `event_type` is not enough to get the necessary information about the configured resources, you can get more information from `client_payload`. For example, you can get the message from the payload in the workfow job step:
```

Comment:
Can you please change this to show how `summary` can be used.

Reply:
I have modified the examples in relation to our discussion in slack. The `summary` field from the webhook payload can be used to define the cluster on which the application is deployed.